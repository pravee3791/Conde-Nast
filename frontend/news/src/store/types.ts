import { Dispatch } from 'redux'


export type Thunk = (
  dispatch: Dispatch<any>,
  getState: () => RootState
) => Promise<any>

export type RootState = ReturnType<any>