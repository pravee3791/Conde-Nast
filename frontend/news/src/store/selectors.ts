
import { RootState } from './store'

export const news = (state: RootState) => state.News;