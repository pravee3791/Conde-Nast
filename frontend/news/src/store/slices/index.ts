import News from "./news";


const reducers = {
    News,
  
}
export default reducers