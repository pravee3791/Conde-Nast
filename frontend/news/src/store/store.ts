
import { configureStore, getDefaultMiddleware } from '@reduxjs/toolkit';
import logger from 'redux-logger'
import { Dispatch } from 'redux'
import appReducer from "./slices/index";


const middlewares = [logger]
const store = configureStore({
    reducer: appReducer,
    middleware: [...getDefaultMiddleware(), ...middlewares]
})

export default store;

// exporting types for TypeScript
export type RootState = ReturnType<typeof store.getState>;
export type Thunk = (
  dispatch: Dispatch<any>,
  getState: () => RootState
) => Promise<any>