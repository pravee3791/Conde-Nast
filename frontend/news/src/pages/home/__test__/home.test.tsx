
import { Provider } from "react-redux";
import {
  BrowserRouter as Router,
} from "react-router-dom";
import Home from '../home';
import { shallow, mount } from 'enzyme';
import Header from "../../../components/header/header";
import Search from "../../../components/search/search";
import NewsItem from "../../../components/newsItem/newsItem";
import store from "../../../store/store";



let wrapped = mount(<Provider store={store}> <Router><Home /></Router></Provider>);
describe('Testing the Home Component', () => {
  test('Home component contains the header comp', () => {
    expect(wrapped.find(Header).length).toEqual(1);
  })

  test('Home component contains the Search comp', () => {
    expect(wrapped.find(Search).length).toEqual(1);
  })

  test('Home component contains the list of news', () => {
    expect(wrapped.find('.home').length).toEqual(1);
  })


  test('Home component shall match the snapshot', () => {
    expect(wrapped).toMatchSnapshot();
  })

})

