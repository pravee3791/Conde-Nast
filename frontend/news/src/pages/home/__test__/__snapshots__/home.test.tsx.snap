// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing the Home Component Home component shall match the snapshot 1`] = `
<Provider
  store={
    Object {
      "@@observable": [Function],
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
   
  <BrowserRouter>
    <Router
      location={
        Object {
          "hash": "",
          "key": "default",
          "pathname": "/",
          "search": "",
          "state": null,
        }
      }
      navigationType="POP"
      navigator={
        Object {
          "action": "POP",
          "back": [Function],
          "block": [Function],
          "createHref": [Function],
          "forward": [Function],
          "go": [Function],
          "listen": [Function],
          "location": Object {
            "hash": "",
            "key": "default",
            "pathname": "/",
            "search": "",
            "state": null,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <Home>
        <Header>
          <div
            className="header"
          >
            <h4 />
          </div>
        </Header>
        <Search>
          <div
            className="wrapper"
          >
            <div
              className="search-container"
            >
              <input
                className="input"
                onChange={[Function]}
                onKeyPress={[Function]}
                placeholder="What are you looking for?"
                type="text"
                value=""
              />
              <input
                className="close-btn"
                onClick={[Function]}
                type="button"
                value="search"
              />
            </div>
          </div>
        </Search>
        <div
          className="home"
        >
          <Loader>
            <div
              className="container"
            >
              <div
                className="card is-loading"
                key="0"
              >
                <div
                  className="image"
                />
                <div
                  className="content"
                >
                  <h2 />
                  <p />
                </div>
              </div>
              <div
                className="card is-loading"
                key="1"
              >
                <div
                  className="image"
                />
                <div
                  className="content"
                >
                  <h2 />
                  <p />
                </div>
              </div>
              <div
                className="card is-loading"
                key="2"
              >
                <div
                  className="image"
                />
                <div
                  className="content"
                >
                  <h2 />
                  <p />
                </div>
              </div>
              <div
                className="card is-loading"
                key="3"
              >
                <div
                  className="image"
                />
                <div
                  className="content"
                >
                  <h2 />
                  <p />
                </div>
              </div>
              <div
                className="card is-loading"
                key="4"
              >
                <div
                  className="image"
                />
                <div
                  className="content"
                >
                  <h2 />
                  <p />
                </div>
              </div>
              <div
                className="card is-loading"
                key="5"
              >
                <div
                  className="image"
                />
                <div
                  className="content"
                >
                  <h2 />
                  <p />
                </div>
              </div>
              <div
                className="card is-loading"
                key="6"
              >
                <div
                  className="image"
                />
                <div
                  className="content"
                >
                  <h2 />
                  <p />
                </div>
              </div>
              <div
                className="card is-loading"
                key="7"
              >
                <div
                  className="image"
                />
                <div
                  className="content"
                >
                  <h2 />
                  <p />
                </div>
              </div>
              <div
                className="card is-loading"
                key="8"
              >
                <div
                  className="image"
                />
                <div
                  className="content"
                >
                  <h2 />
                  <p />
                </div>
              </div>
              <div
                className="card is-loading"
                key="9"
              >
                <div
                  className="image"
                />
                <div
                  className="content"
                >
                  <h2 />
                  <p />
                </div>
              </div>
            </div>
          </Loader>
          <div
            className="container"
          />
        </div>
      </Home>
    </Router>
  </BrowserRouter>
</Provider>
`;
